rpcs3/Emu/Cell/lv2/sys_sync.h

```diff
@@ -218,6 +218,10 @@ struct lv2_obj
 	template<bool is_usleep = false>
 	static bool wait_timeout(u64 usec, cpu_thread* const cpu = nullptr)
 	{
+		// Clamp to (1<<48) - 1 max as real ps3, this also solves potential overflows when scaling
+		if (usec > (1ull << 48) - 1) usec = (1ull << 48) - 1;
```

Comment:
Don't pay too much attention to 2^48-1 constant from the kernel. It has no relevance for emulator code (just a lv2 implementation detail), because it roughly corresponds to 9 years. It's more useful to clamp it to max_timeout from the start.